scripts/scil_denoise_sf.py

```diff
@@ -0,0 +1,122 @@
+
+from scilpy.denoise.asym_enhancement import local_asym_gaussian_filtering
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('--out_dir', default='.',
+                   help='Output directory. Default is current directory.')
+
+    p.add_argument('--out_sh', default='out_filtered.nii.gz',
+                   help='File name for averaged signal. [%(default)s]')
+
+    p.add_argument('--out_mask', default='mask.nii.gz',
+                   help='File name for output mask. [%(default)s]')
+
+    p.add_argument('--mask_eps', default=1e-16,
+                   help='Threshold on SH coefficients norm for output mask.'
+                   ' [%(default)s]')
+
+    p.add_argument(
+        '--sh_order', default=8, type=int,
+        help='SH order of the input. [%(default)s]')
+
+    p.add_argument(
+        '--sphere', default='repulsion724', type=str,
+        help='Sphere used for the SH projection. [%(default)s]'
+    )
+
+    p.add_argument(
+        '--sharpness', default=1.0, type=float,
+        help='Specify sharpness factor to use for weighted average.'
+        ' [%(default)s]'
+    )
+
+    p.add_argument(
+        '--sigma', default=1.0, type=float,
+        help='Sigma of the gaussian to use. [%(default)s]'
+    )
+
+    p.add_argument(
+        '--out_sym', action='store_true',
+        help='Save output in symmetric SH basis.'
+    )
+
+    add_sh_basis_args(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def generate_mask(sh, threshold):
+    norm = np.linalg.norm(sh, axis=-1)
+    mask = norm > threshold
+    return mask
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.basicConfig(level=logging.INFO)
+
+    # Checking args
+    if not os.path.exists(args.out_dir):
+        os.makedirs(args.out_dir)
+    out_sh = os.path.join(args.out_dir, args.out_sh)
+    out_mask = os.path.join(args.out_dir, args.out_mask)
+
+    outputs = [out_sh, out_mask]
+    assert_inputs_exist(parser, args.in_sh)
+    assert_outputs_exist(parser, args, outputs)
+
+    # Prepare data
+    sh_img = nib.nifti1.load(args.in_sh)
+    data = sh_img.get_fdata(dtype=np.float)
+
+    logging.info('Executing locally asymmetric Gaussian filtering.')
+    filtered_sh = local_asym_gaussian_filtering(
+        data, sh_order=args.sh_order,
+        sh_basis=args.sh_basis,
+        out_full_basis=not(args.out_sym),
+        sphere_str=args.sphere,
+        dot_sharpness=args.sharpness,
+        sigma=args.sigma)
+
+    nib.save(nib.Nifti1Image(filtered_sh.astype(np.float), sh_img.affine),
+             out_sh)
+
+    # Generate mask by applying threshold on input SH amplitude
+    mask = generate_mask(data, args.mask_eps)
+    nib.save(nib.Nifti1Image(mask.astype(np.uint8), sh_img.affine),
```

Comment:
Same here, generate_mask should cast as uint8 all the time, right? Its called generate mask, so I would expect it returns a mask as the right type already

Reply:
`generate_mask` returns a Boolean array, which is then saved as an Integer array. Wouldn't it be weird for `generate_mask` to return the mask as an array of unsigned int, since I get the feeling a mask should be `bool` rather than `int`?